spec/rubocop/cop/rspec/unspecified_exception_spec.rb

```diff
@@ -171,5 +171,14 @@
         }.to raise_exception(my_exception)
       RUBY
     end
+
+    it 'detects chained offenses' do
+      expect_offense(<<~RUBY)
+        expect {
+          foo
+        }.to raise_exception.and change { bar }
```

Comment:
What if it’s `raise_error().and change {}.and change {}`?

Reply:
In the end, I decided to radically change the implementation to support the cases I commented on.
Instead of searching from `to` node, I will change it to search from `raise_exception` or `raise_error` nodes. I think this will eliminate the possibility of false-negatives and make it easier to report error locations.